app/controllers/accounts_controller.rb

```diff
@@ -2,29 +2,18 @@ class AccountsController < ApplicationController
   before_action :authenticate_user!
 
   def new
-    if params[:type].blank? || Account.accountable_types.include?("Account::#{params[:type]}")
-      @account = if params[:type].blank?
-        Account.new
-      else
-        Account.new(accountable_type: "Account::#{params[:type]}")
-      end
-    else
-      head :not_found
-    end
+    @account = build_account
+    head :not_found unless @account
```

Comment:
I would advise against using modifier clauses like this.  Standard Rails is to not use them in controller actions

For example here - a user will at first see that it returns `:not_found`.  Its not until the end of the line they realise its conditional.

I know this is subjective and both ways work and it saves a couple of lines, but for readability and consistency with other controller actions we should defer to the more verbose if...else...

Reply:
I agree it is subjective. And I buy your point regarding consistencies with other controllers, which is rather also subjective in my opinion. 

Perhaps we can decide what things should look like but I don't think that "rhyme" is totally given in a large codebase mostly with `if..else` conditional statements or tennary operators or modifier clause. That uniformity is not given that we'll have only `if..else` statements accross the board. It is not achievable!

On the other hand, if I take it a bit further to avert concerns in your example and pass in some `..else` statement, this is what I think:

```
def new
  @account = build_account

  if @account.blank?
    head :not_found
  end
end
```

```
def create
  @account = build_and_associate_account

  unless @account.save
    render "new", status: :unprocessable_entity
  else
    redirect_to accounts_path, notice: "New account created successfully"
  end
end
```

Afterall, it's Ruby under the hood with dashes of Rails additions. So we don't have to follow Rails standards in such a situation. I really just want to clean these controllers one after the other and set some precedence we follow in this repo. Let me know what you think using the `@account`, but I certainly take preference in what I provided and I'll accept it in any repository since it works.

Reply:
Absolutely not seeing the benefits too. Just trying to regularize the uniformity in your opinion.